src/compiler/checker.ts

```diff
@@ -1357,10 +1359,19 @@ namespace ts {
             }
 
             const container = getEnclosingBlockScopeContainer(declaration);
-            return !!(usage.flags & NodeFlags.JSDoc)
-                || isInTypeQuery(usage)
-                || isUsedInFunctionOrInstanceProperty(usage, declaration, container)
-                   && !(compilerOptions.target === ScriptTarget.ESNext && !!compilerOptions.useDefineForClassFields);
+            if (!!(usage.flags & NodeFlags.JSDoc) || isInTypeQuery(usage)) {

```

Comment:
I bet this condition actually could be the same as `isValidTypeOnlyAliasUseSite`. But probably not worth changing.